tracer/src/Datadog.Trace/PDBs/SourceLinkInformationExtractor.cs

```diff
@@ -0,0 +1,124 @@
+        // If these conditions weren't met, the attributes won't be there, so we'll need to extract the information from the PDB file.
+
+        return ExtractFromAssemblyAttributes(assembly, out commitSha, out repositoryUrl) ||
+               ExtractFromPdb(assembly, out commitSha, out repositoryUrl);
+    }
+
+    private static bool ExtractFromPdb(Assembly assembly, [NotNullWhen(true)] out string? commitSha, [NotNullWhen(true)] out string? repositoryUrl)
+    {
+        commitSha = null;
+        repositoryUrl = null;
+
+        var pdbReader = DatadogPdbReader.CreatePdbReader(assembly);
+        if (pdbReader == null)
+        {
+            Log.Information("PDB file for assembly {AssemblyFullPath} could not be found", assembly.Location);
+            return false;
+        }
+
+        var sourceLinkJsonDocument = pdbReader.GetSourceLinkJsonDocument();
+        if (sourceLinkJsonDocument == null)
+        {
+            Log.Information("PDB file {PdbFullPath} does not contain SourceLink information", pdbReader.PdbFullPath);
+            return false;
+        }
+
+        if (!ExtractSourceLinkMappingUrl(sourceLinkJsonDocument, pdbReader.PdbFullPath, out var sourceLinkMappedUri))
+        {
+            return false;
+        }
+
+        return CompositeSourceLinkUrlParser.Instance.TryParseSourceLinkUrl(sourceLinkMappedUri, out commitSha, out repositoryUrl);
+    }
+
+    // Grab the SourceLink mapping URL. For example,
+    // From:
+    //       {"documents":{"C:\\dev\\dd-trace-dotnet\\*":"https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/dd35903c688a74b62d1c6a9e4f41371c65704db8/*"}}
+    // Extract:
+    //       https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/dd35903c688a74b62d1c6a9e4f41371c65704db8/*
+    private static bool ExtractSourceLinkMappingUrl(string sourceLinkJsonDocument, string pdbFullPath, [NotNullWhen(true)] out Uri? sourceLinkMappedUri)
+    {
+        sourceLinkMappedUri = null;
+        try
+        {
+            string? sourceLinkMappedUrl = JObject.Parse(sourceLinkJsonDocument).SelectTokens("$.documents.*").FirstOrDefault()?.ToString();
+            if (string.IsNullOrWhiteSpace(sourceLinkMappedUrl) || !Uri.TryCreate(sourceLinkMappedUrl, UriKind.Absolute, out sourceLinkMappedUri))
+            {
+                Log.Information("PDB file {PdbFullPath} contained SourceLink information, but we failed to parse it.", pdbFullPath);
+                return false;
+            }
+
+            Log.Information("PDB file {PdbFullPath} contained SourceLink information, and we successfully parsed it. The mapping uri is {SourceLinkMappedUri}.", pdbFullPath, sourceLinkMappedUri);
+            return true;
+        }
+        catch (Exception e)
+        {
+            Log.Warning(e, "PDB file {PdbFullPath} contained SourceLink document {document}, but we failed to parse it.", pdbFullPath, sourceLinkJsonDocument);
```

Comment:
tiny nit: new analyzers enforce pascal case in logger properties
```suggestion
            Log.Warning(e, "PDB file {PdbFullPath} contained SourceLink document {Document}, but we failed to parse it.", pdbFullPath, sourceLinkJsonDocument);
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/commit/989c9c823ebb86da47f43eeaf48a2466a04e38af